### PR TITLE
Update documentation

### PR DESCRIPTION
**Pull request by unlucio**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,36 +2,121 @@
 
 [![Build Status](https://travis-ci.org/namshi/reconfig.svg?branch=travis)](https://travis-ci.org/namshi/reconfig)
 
-Handy and _strongly_ opinionated config helper on top of [Reconfig](https://github.com/namshi/reconfig).
+Handy and _strongly_ opinionated config helper on top of [reconfig](https://github.com/namshi/reconfig).
 
-## Installation
+## So what about it?
+Here in Namshi we like our configuration incremental, over-ridable, and defined in a compact way. As well as we like to don't go nuts when we clone a project repo for the 1st time.
```
Comment (cirpo) [inline README.md]: overridable?

Comment (cirpo) [inline README.md]: Here in Namshi we like our configuration TO BE?

File: README.md [hunk 2]
```diff
@@ -2,36 +2,121 @@
...
+This is way we came up with tools like [reconfig](https://github.com/namshi/reconfig) and [file-ensure](https://github.com/namshi/node-file-ensure). With nmConfig we put all this things together in a convenient lib that turns all of this in a matter of a `require()` instruction.
```
Comment (cirpo) [inline README.md:10]: all these

File: README.md (hunk 1 above)
Comment (unlucio) [inline README.md]: maybe `:`

File: README.md [hunk 3]
```diff
@@ -2,36 +2,121 @@
...
+- Automatically creates a `config/` directory and the needed files if are not present:
+
+```
+config/
+     |- base.yml
+     |- dev.example.yml
+     |- staging.yml
+     |- live.yml
 ```
+- Compiles your config files, figure out your environment, and applies the needed merging on the `base.yml` one:
```
Comment (cirpo) [inline README.md]: your environment and applies <- no comma

File: README.md [hunk 4]
```diff
@@ -2,36 +2,121 @@
...
-This library is tested through mocha, simply run either
+- returns you a `reconfig` instance:
+```javascript
+console.log(config.get('characters.anakin'));
 
+// ==> 'sith'
 ```
-mocha
+
+- Figures out a `recofig`'s env overrider prefix from your `package.json`:
```
Comment (cirpo) [inline README.md:58]: overridden?

File: README.md [hunk 5]
```diff
@@ -2,36 +2,121 @@
...
+{
+  "name": "myApp",
+  "version": "0.1.0",
+  "description": "this is my app, there are many like it, but this one is mine!"
+// [...]
+}
 ```
 
-or
+will yeld a `MYAPP_CONFIG` env prefix for `reconfig` (check [this section](https://github.com/namshi/reconfig#nodejs-specifics) on `reconfig`'s doc for more infos on what this does)
```
Comment (cirpo) [inline README.md]: yield?

on what it does?

File: README.md [hunk 6]
```diff
@@ -2,36 +2,122 @@
...
+Here in Namshi we like our configuration: incremental, overridable, and defined in a compact way.<br/>
+As well as we like to don't go nuts when we clone a project repo for the 1st time.<br/>
+This is way we came up with tools like [reconfig](https://github.com/namshi/reconfig) and [file-ensure](https://github.com/namshi/node-file-ensure). With nmConfig we put all this things together in a convenient lib that turns all of this in a matter of a `require()` instruction.
 
-Install this library via [NPM](https://www.npmjs.org/package/node-nmconfig):
+## What does it do:
 
-``` bash
-npm install node-nmconfig
+- Automatically creates a `config/` directory and the needed files if are not present:
```
Comment (odino) [inline README.md]: missing `they`

File: README.md (hunk 3 above)
Comment (odino) [inline README.md]: figure**s**

**Comment by odino** [comment]
:+1:

File: README.md (hunk 4 above)
Comment (unlucio) [inline README.md:58]: nope, that's another thing.
It's reconfig thing: overlays coming form the env overriding config values

File: README.md [hunk 7]
```diff
@@ -2,36 +2,121 @@
...
 ```
 
-if you don't have mocha installed globally.
+If you need it on the client side we highly recommend
+[browserify](http://browserify.org/).
+
+## Tests
+This library is a little convenience wrapper on top of extensively tested projects,
+so for once we can be a little bit lazy and skip them ;)
+We do like tests tho, so if you feel giving us a hand we'll be more than happy
```
Comment (cirpo) [inline README.md:122]: mhhhhh

File: README.md (hunk 5 above)
Comment (unlucio) [inline README.md]: will cause: 
`new reconfig(configValues, 'MYAPP_CONFIG');`
that's why I referred the specific part to understand what it does.

File: README.md (hunk 7 above)
Comment (unlucio) [inline README.md:122]: send me the tests :D

**Comment by unlucio** [comment]
k, thanks for the review :D 
fixed my traditional typos. Merging it :)

**Pull request merged**